catalog/update.go

```diff
@@ -0,0 +1,40 @@
+package catalog
+
+import (
+	"context"
+
+	"github.com/databrickslabs/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func updateFunctionFactory(securable string, updatable []string) func(context.Context, *schema.ResourceData, *common.DatabricksClient) error {
+	return func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+		patch := map[string]interface{}{}
+		for _, field := range updatable {
+			_, new := d.GetChange(field)
+			if !d.HasChange(field) {
+				continue
+			}
+			patch[field] = new
+		}
+		if len(patch) == 0 {
+			return nil
+		}
+		switch securable {
+		case "metastore":
+			return NewMetastoresAPI(ctx, c).updateMetastore(d.Id(), patch)
+		case "catalog":
+			return NewCatalogsAPI(ctx, c).updateCatalog(d.Id(), patch)
+		case "schema":
+			return NewSchemasAPI(ctx, c).updateSchema(d.Id(), patch)
+		case "table":
+			return NewTablesAPI(ctx, c).updateTable(d.Id(), patch)
+		case "external-location":
+			return NewExternalLocationsAPI(ctx, c).update(d.Id(), patch)
+		case "storage-credential":
+			return NewStorageCredentialsAPI(ctx, c).update(d.Id(), patch)
+		default:
+			return nil
```

Comment:
must be an error here ;) like "securable is not implemented for updates" or something. with "layer leak" this is not really a problem